packages/esm-outpatient-app/src/patient-search/visit-form/visit-form.component.tsx

```diff
@@ -44,21 +53,80 @@ const StartVisitForm: React.FC<VisitFormProps> = ({ patientUuid, toggleSearchTyp
   const [isMissingVisitType, setIsMissingVisitType] = useState(false);
   const [isSubmitting, setIsSubmitting] = useState(false);
   const [selectedLocation, setSelectedLocation] = useState('');
-  const [timeFormat, setTimeFormat] = useState(new Date().getHours() >= 12 ? 'PM' : 'AM');
+  const [timeFormat, setTimeFormat] = useState<amPm>(new Date().getHours() >= 12 ? 'PM' : 'AM');
   const [visitDate, setVisitDate] = useState(new Date());
   const [visitTime, setVisitTime] = useState(dayjs(new Date()).format('hh:mm'));
   const [visitType, setVisitType] = useState<string | null>(null);
   const state = useMemo(() => ({ patientUuid }), [patientUuid]);
   const allVisitTypes = useVisitTypes();
+  const [ignoreChanges, setIgnoreChanges] = useState(true);
 
   useEffect(() => {
     if (locations && sessionUser?.sessionLocation?.uuid) {
       setSelectedLocation(sessionUser?.sessionLocation?.uuid);
     }
   }, [locations, sessionUser]);
 
+  const handleSubmit = useCallback(
+    (event) => {
+      event.preventDefault();
+
+      if (!visitType) {
+        setIsMissingVisitType(true);
+        return;
+      }
+
+      setIsSubmitting(true);
+
+      const [hours, minutes] = convertTime12to24(visitTime, timeFormat);
+
+      const payload: NewVisitPayload = {
+        patient: patientUuid,
+        startDatetime: toDateObjectStrict(
+          toOmrsIsoString(
+            new Date(dayjs(visitDate).year(), dayjs(visitDate).month(), dayjs(visitDate).date(), hours, minutes),
+          ),
+        ),
+        visitType: visitType,
+        location: selectedLocation,
+      };
+
+      const abortController = new AbortController();
+      saveVisit(payload, abortController)
+        .pipe(first())
+        .subscribe(
+          (response) => {
+            if (response.status === 201) {
+              showToast({
+                kind: 'success',
+                description: t(
+                  'startVisitSuccessfully',
+                  'Visit started successfully. Patient has been added to active visits list.',
+                  `${hours} : ${minutes}`,
+                ),
+              });
+              navigate({ to: `\${openmrsSpaBase}/home` });
```

Comment:
@CynthiaKamau when you start visit on outpatient app, would still want to navigate out of the app or just to close that workspace? not so sure about the navigation path here.

Reply:
updating this